### DISCUSSION TITLE
Appending to a relation in Go

### DISCUSSION THREAD

**Discussion by ndarilek** [pocketbase/pocketbase#4460] [Q&A] [answered]
Hey, having a hard time piecing this together from the docs. I have an images collection, a followups collection, and an image can have multiple followups. How do I work with the followups relation in Go?

The only way I can think of is just setting it to an array of records each time, but that seems like it might not be correct. Is that the only way or am I missing something?

Thanks a bunch.

**Comment by ganigeorgiev** [comment, marked as answer]
Multiple relations are stored as an array of ids, so to append new value you can fetch the existing one, add your new id(s) and then set back the updated slice as a new field value.

```go
rels := record.GetStringSlice("yourRelField")

rels = append(rels, "newId1", "newId2")

record.Set("yourRelField", rels)

app.Dao().SaveRecord(record)
```

**Reply by ganigeorgiev** [reply]
I forgot to mention that the above doesn't perform validations (it is a direct db insert/update).
If you want to perform validations (eg. whether the correct relation ids are assigned) at the moment you'll have to create a `forms.RecordUpsert` - https://pocketbase.io/docs/go-records/#update-record-with-data-validations. Since usually the forms are intended to be populated from request data, it also supports the [field suffix `+`/`-` modifiers](https://pocketbase.io/docs/working-with-relations/#append-to-multiple-relation) to add/remove relation entries:
```go
form.LoadData(map[string]any{
    // to add entries
    "yourRelField+": []string{"newId1", "newId2"},
    // to remove entries
    "yourRelField-": []string{"oldId1"},
})
```

Note that this will simplified as part of the [planned refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and the form creation will no longer be necessary since we'll run the validation by default on Save.

**Reply by ndarilek** [reply]
Awesome, thank you!
